File: frontend/src/services/invoiceItemService.js
import invoiceApi from "../api/invoiceItemApi.js";

const invoiceItemService = {
    async addItem(invoiceNumber, productId) {
        const response = await invoiceApi.addItem(invoiceNumber, productId);
        return response.data;
    },

    async getItems(invoiceNumber) {
        const response = await invoiceApi.getItems(invoiceNumber);
        return response.data;
    },

    async updateQtyItem(invoiceItemId, qty) {
        const response = await invoiceApi.updateQtyItem(invoiceItemId, qty);
        return response.data;
    },

    async updateSellingPrice(invoiceItemId, sellingPrice) {
        const response = await invoiceApi.updateSellingPrice(
            invoiceItemId,
            sellingPrice
        );
        return response.data;
    },

    async deleteItem(invoiceItemId) {
        const response = await invoiceApi.deleteItem(invoiceItemId);
        return response.data;
    },
};

export default invoiceItemService;

File: backend/src/main/java/com/sr/mart/software/service/InvoiceItemService.java
package com.sr.mart.software.service;

import com.sr.mart.software.model.InvoiceItemResponse;
import java.math.BigDecimal;
import java.util.List;

public interface InvoiceItemService {

    InvoiceItemResponse createInvoiceLineItems(String addInvoiceLineItems, Long productId);

    List<InvoiceItemResponse> getInvoiceLineItems(String invoiceNumber);

    InvoiceItemResponse updateQty(Long invoiceItemId, Double qty);

    InvoiceItemResponse updateSellingPrice(Long invoiceItemId, BigDecimal sellingPrice);

    void deleteInvoice(Long invoiceItemId);
}
